### PR TITLE
Eggnogmapper

### PR DESCRIPTION
**Pull request by vagkaratzas**

### CODE REVIEW

File: modules/ebi-metagenomics/eggnogmapper/main.nf [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        --data_dir ${eggnog_data_dir} \\
+        -m diamond \\
+        --dmnd_db ${eggnog_diamond_db} \\
+        --database ${eggnog_db} \\
+        --output ${prefix} \\
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        eggnog-mapper: $VERSION
```
Comment (mberacochea) [inline modules/ebi-metagenomics/eggnogmapper/main.nf:40]: It's better to capture the version from the emapper.py script directly:
```
\$(echo \$(emapper.py --version) | grep -o "emapper-[0-9]\\+\\.[0-9]\\+\\.[0-9]\\+" | sed "s/emapper-//")
```
this expression should do the trick

File: tests/modules/ebi-metagenomics/eggnogmapper/test.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+- name: eggnogmapper test_eggnogmapper
+  command: nextflow run ./tests/modules/ebi-metagenomics/eggnogmapper -entry test_eggnogmapper -c ./tests/config/nextflow.config -c ./tests/modules/ebi-metagenomics/eggnogmapper/nextflow.config
+  tags:
+    - eggnogmapper
+  files:
+    - path: output/eggnogmapper/test.emapper.hits
+      md5sum: f8629819c48c888275dbb1a09749c891
+    - path: output/eggnogmapper/versions.yml
```
Comment (mberacochea) [inline tests/modules/ebi-metagenomics/eggnogmapper/test.yml:8]: You are missing the md5sum of the `versions.yml` file

File: modules/ebi-metagenomics/eggnogmapper/main.nf [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = '2.1.12'
+    """
+    emapper.py \\
+        --cpu ${task.cpus} \\
+        -i ${fasta} \\
```
Comment (mberacochea) [inline modules/ebi-metagenomics/eggnogmapper/main.nf:30]: Add support got .gz fasta files please

**Review by mberacochea — CHANGES_REQUESTED** [review]
I've added some comments. Thank you @vagkaratzas, this is great stuff

File: modules/ebi-metagenomics/eggnogmapper/main.nf (hunk 3 above)
Comment (vagkaratzas) [reply modules/ebi-metagenomics/eggnogmapper/main.nf:30]: In the meta.yml file the fasta pattern is: pattern: "*.{fasta,fa,fasta.gz,fa.gz}"
Should work fine, right?

**Review by mberacochea — APPROVED** [review]
:shipit:

**Pull request merged**